File: mobile/ios/TurkuazTelefonRehberiIOS/config/MobileConfig.swift
// # 📄 Dosya Yolu: C:/Projects/TelefonRehberi/mobile/ios/TurkuazTelefonRehberiIOS/config/MobileConfig.swift
// # 📌 Amac: iOS mobil istemcinin API ve kalici ayar sabitlerini merkezi olarak tutar.
// # 📌 Config - Swift
// # Version: 2.37.0
// # Aciklama: Genisletilmis contact form alanlari, endpoint, header, HTTP ve UserDefaults sabitlerini tanimlar.
// # Bagimli Oldugu Katman: Config
import Foundation

enum MobileConfig {
    static let preferencesServerUrl = "server_url"
    static let preferencesTokenLegacy = "sync_token"
    static let preferencesSyncMapNativePrefix = "sync_map_native_"
    static let preferencesSyncMapUuidPrefix = "sync_map_uuid_"
    static let keychainService = "com.turkuazlabs.telefonrehberi.sync"
    static let keychainTokenAccount = "sync_token"
    static let preferencesDeviceId = "device_id"
    static let statusPath = "/api/v1/status"
    static let contactsPath = "/api/v1/contacts"
    static let importPath = "/api/v1/import"
    static let authorizationHeader = "Authorization"
    static let bearerPrefix = "Bearer "
    static let contentTypeHeader = "Content-Type"
    static let contentTypeForm = "application/x-www-form-urlencoded; charset=UTF-8"
    static let acceptHeader = "Accept"
    static let contentTypeJson = "application/json"
    static let formName = "name"
    static let formPhone = "phone"
    static let formPhoneSecondary = "phone_secondary"
    static let formPhoneWork = "phone_work"
    static let formEmail = "email"
    static let formEmailSecondary = "email_secondary"
    static let formCompany = "company"
    static let formJobTitle = "job_title"
    static let formBirthday = "birthday"
    static let formWebsite = "website"
    static let formCategory = "category"
    static let formAddress = "address"
    static let formCity = "city"
    static let formDistrict = "district"
    static let formPostalCode = "postal_code"
    static let formCountry = "country"
    static let formNotes = "notes"
    static let formFavorite = "favorite"
    static let formPhones = "phones"
    static let formEmails = "emails"
    static let formPhotoBase64 = "photo_base64"
    static let formDeviceId = "device_id"
    static let formExternalId = "external_id"
    static let formSyncUuid = "sync_uuid"
    static let httpGet = "GET"
    static let httpPost = "POST"
    static let httpSuccessRange = 200..<300
    static let methodRowSeparator = "~"
    static let methodFieldSeparator = "|"
    static let methodPrimaryTrue = "1"
    static let methodPrimaryFalse = "0"
    static let methodComponentAllowedCharacters = CharacterSet.alphanumerics.union(CharacterSet(charactersIn: "-_."))
    static let requestTimeout: TimeInterval = 20
}

File: mobile/android/app/src/main/java/com/turkuazlabs/telefonrehberi/mobile/config/MobileConfig.java
// # 📄 Dosya Yolu: C:/Projects/TelefonRehberi/mobile/android/app/src/main/java/com/turkuazlabs/telefonrehberi/mobile/config/MobileConfig.java
// # 📌 Amac: Android mobil istemcinin teknik sabitlerini merkezi olarak tutar.
// # 📌 Config - Java
// # Version: 2.37.0
// # Aciklama: Genisletilmis API JSON/form alanlari, endpoint, header, HTTP, preference ve izin sabitlerini tanimlar.
// # Bagimli Oldugu Katman: Config
package com.turkuazlabs.telefonrehberi.mobile.config;

public final class MobileConfig {
    public static final String PREFERENCES_NAME = "turkuaz_phonebook_mobile";
    public static final String PREF_SERVER_URL = "server_url";
    public static final String PREF_TOKEN_LEGACY = "sync_token";
    public static final String PREF_TOKEN_ENCRYPTED = "sync_token_encrypted";
    public static final String PREF_TOKEN_IV = "sync_token_iv";
    public static final String PREF_SYNC_MAP_NATIVE_PREFIX = "sync_map_native_";
    public static final String PREF_SYNC_MAP_UUID_PREFIX = "sync_map_uuid_";
    public static final String KEYSTORE_TOKEN_ALIAS = "turkuaz_phonebook_sync_token";
    public static final String PREF_DEVICE_ID = "device_id";
    public static final String STATUS_PATH = "/api/v1/status";
    public static final String CONTACTS_PATH = "/api/v1/contacts";
    public static final String IMPORT_PATH = "/api/v1/import";
    public static final String AUTHORIZATION_HEADER = "Authorization";
    public static final String BEARER_PREFIX = "Bearer ";
    public static final String CONTENT_TYPE_HEADER = "Content-Type";
    public static final String CONTENT_TYPE_FORM = "application/x-www-form-urlencoded; charset=UTF-8";
    public static final String ACCEPT_HEADER = "Accept";
    public static final String CONTENT_TYPE_JSON = "application/json";
    public static final String JSON_CONTACTS = "contacts";
    public static final String JSON_ID = "id";
    public static final String FIELD_SYNC_UUID = "sync_uuid";
    public static final String FIELD_NAME = "name";
    public static final String FIELD_PHONE = "phone";
    public static final String FIELD_PHONE_SECONDARY = "phone_secondary";
    public static final String FIELD_PHONE_WORK = "phone_work";
    public static final String FIELD_EMAIL = "email";
    public static final String FIELD_EMAIL_SECONDARY = "email_secondary";
    public static final String FIELD_COMPANY = "company";
    public static final String FIELD_JOB_TITLE = "job_title";
    public static final String FIELD_BIRTHDAY = "birthday";
    public static final String FIELD_WEBSITE = "website";
    public static final String FIELD_CATEGORY = "category";
    public static final String FIELD_ADDRESS = "address";
    public static final String FIELD_CITY = "city";
    public static final String FIELD_DISTRICT = "district";
    public static final String FIELD_POSTAL_CODE = "postal_code";
    public static final String FIELD_COUNTRY = "country";
    public static final String FIELD_NOTES = "notes";
    public static final String FIELD_FAVORITE = "favorite";
    public static final String FIELD_PHONES = "phones";
    public static final String FIELD_EMAILS = "emails";
    public static final String FIELD_PHOTO_BASE64 = "photo_base64";
    public static final String FORM_DEVICE_ID = "device_id";
    public static final String FORM_EXTERNAL_ID = "external_id";
    public static final String FORM_SYNC_UUID = "sync_uuid";
    public static final String HTTP_GET = "GET";
    public static final String HTTP_POST = "POST";
    public static final int HTTP_SUCCESS_MIN = 200;
    public static final int HTTP_SUCCESS_MAX_EXCLUSIVE = 300;
    public static final int HTTP_CONNECT_TIMEOUT_MS = 10000;
    public static final int HTTP_READ_TIMEOUT_MS = 20000;
    public static final int PERMISSION_REQUEST_CONTACTS = 1001;

    private MobileConfig() {
    }
}
